summingbird-batch/src/main/scala/com/twitter/summingbird/batch/BatchID.scala

```diff
@@ -124,22 +123,21 @@ object BatchID {
 
   implicit val batchID2Bytes: Injection[BatchID, Array[Byte]] =
     Injection.connect[BatchID, Long, Array[Byte]]
+
+  implicit val ord: Ordering[BatchID] = Ordering.by(_.id)
 }
 
-class BatchID(val id: Long) extends Ordered[BatchID] with java.io.Serializable {
+case class BatchID(id: Long) extends Ordered[BatchID] with java.io.Serializable {
   def next: BatchID = new BatchID(id + 1)
   def prev: BatchID = new BatchID(id - 1)
   def +(cnt: Long) = new BatchID(id + cnt)
   def -(cnt: Long) = new BatchID(id - cnt)
+
   def compare(b: BatchID) = id.compareTo(b.id)
+
   def max(b: BatchID) = if (this >= b) this else b
   def min(b: BatchID) = if (this < b) this else b
-  override lazy val toString = "BatchID." + id.toString
 
-  override def hashCode: Int = id.hashCode
-  override def equals(other: Any): Boolean =
-    other match {
-      case that: BatchID => this.id == that.id
-      case _ => false
-    }
+  override lazy val toString = "BatchID." + id.toString
+  override lazy val hashCode: Int = id.hashCode
```

Comment:
does this need to be a lazy val? Maybe just val and not have the thread sync costs? hashCode of a Long must be really fast.